### PR TITLE
Vaccination feature

### LINKED ISSUE #48: Vaccination structure
Structure includes the following class/interface:
1. `Vaccination` - to represent a vaccination
2. `VaccinationChecker` - to check if a person (patient) is able to take a the shot
    - To avoid cyclic dependency
4. `VaccinationRecord` - to contain a vaccination record; the vaccination and the date it was taken

### PR DESCRIPTION
**Pull request by daitenshionyan**
Resolves #48.

Suggestion:
Move `VaxChecker` logic to appointment.

`VaxChecker` only evaluates if the patient can take the vaccination or not but does not provide reasons as to why they are unable to.

Integrating its logic into the addition of new appoints and editing a bit such that it throws custom exceptions when the patient cannot take the vaccination might solve this.

### CODE REVIEW

File: src/main/java/seedu/vms/model/vaccination/Vaccination.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
+package seedu.vms.model.vaccination;
+
+import java.util.List;
+
+
+/** Represents a vaccination. */
+public enum Vaccination {
```
Comment (francisyzy) [inline src/main/java/seedu/vms/model/vaccination/Vaccination.java:7]: Are we going to hard code the vaccine types inside the code or should we put it in a `.json` file?

File: src/main/java/seedu/vms/model/vaccination/DoseType.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package seedu.vms.model.vaccination;
+
+
+/** Represents the type of a dose. */
+public enum DoseType {
```
Comment (francisyzy) [inline src/main/java/seedu/vms/model/vaccination/DoseType.java:5]: Will there be any chance of more than these 4 doses?

Comment (daitenshionyan) [reply src/main/java/seedu/vms/model/vaccination/DoseType.java]: Have edited to accept as many does types (no rename to group) as the computer can handle.

**Review by francisyzy — APPROVED** [review]
lgtm

**Pull request merged**